### ISSUE TITLE
Breaking changes in ipywidgets v8 and jupyterlab-widgets v3

### ISSUE THREAD

**Issue by flekschas** [jupyter-widgets/ipywidgets#3607]
## Description

I am the developer of a [jupyter widgets](https://github.com/flekschas/jupyter-scatter) and I noticed that my widget started to fail with ipywidgets v8 and/or jupyterlab-widgets v3. (What happened to jupyterlab-widgets v2 by the way?)

The error message is:

```
Model class 'JupyterScatterModel' from module 'jupyter-scatter' is loaded but can not be instantiated
```

I looked over the [ipywidgets v8 changelog](https://github.com/jupyter-widgets/ipywidgets/releases/tag/8.0.0) but couldn't find anything mentioning of breaking changes. However, one or multiple changes must cause the failing instantiation.

## Reproduce

1. Create the following `environment-py39-ipyw8-jlab3.yml`

```
name: jscatter-py39-ipyw8-jlab3

channels:
  - conda-forge
  - defaults

dependencies:
  - python==3.9.*
  - nodejs==16.*
  - numpy
  - pandas
  - pip
  - pip:
    - jupyterlab==3.*
    - ipywidgets==8.*
    - jupyterlab_widgets==3.*
    - jupyter-scatter
```

2. Then do

```
conda env create -f ./environment-py39-ipyw8-jlab3.yml 
conda activate jscatter-py39-ipyw8-jlab3
jupyter-lab
```

3. Then run

```
import numpy as np
import jscatter
jscatter.plot(x=np.random.rand(500), y=np.random.rand(500))
```

The widget will fail to instantiate.

For comparison, create the `environment-py39-ipyw7-jlab1.yml` and repeat steps two and three.

```
name: jscatter-py39-ipyw7-jlab1

channels:
  - conda-forge
  - defaults

dependencies:
  - python==3.9.*
  - nodejs==18.*
  - numpy
  - pandas
  - pip
  - pip:
    - jupyterlab==3.*
    - ipywidgets==7.*
    - jupyterlab_widgets==1.*
    - jupyter-scatter
```

This time the widget will load as expected.

## Expected behavior

The widget should load as expected or at the very least it should throw a useful error message.

### If using JupyterLab
- JupyterLab version: 3.4.8

<details><summary>Installed Labextensions</summary>
<pre>
pyterLab v3.4.8
/Users/me/miniforge3/envs/jscatter-py39-ipyw8-jlab3/share/jupyter/labextensions
        jupyterlab_pygments v0.2.2 enabled OK (python, jupyterlab_pygments)
        jupyter-scatter v0.7.1 enabled OK
        @jupyter-widgets/jupyterlab-manager v5.0.3 enabled OK (python, jupyterlab_widgets)
</pre>
</details>

**Comment by flekschas** [comment]
The one thing I see in the web dev console which might hint at the issue is the following error message:

```
r._deserialize_state is not a function
```

This makes me wonder if the issue is somehow related to https://github.com/jupyter-widgets/ipywidgets/pull/3549

**Comment by martinRenou** [comment]
Thanks for opening an issue. Did you follow the migration guide for your library? https://ipywidgets.readthedocs.io/en/stable/migration_guides.html#migrating-from-7-x-to-8-0

**Comment by martinRenou** [comment]
I see you are [using the old `.extend` method](https://github.com/flekschas/jupyter-scatter/blob/master/js/src/index.js#L14), you will need to modernize your classes definitions. For example [ipysheet did that some days ago](https://github.com/QuantStack/ipysheet/pull/233).

**Comment by flekschas** [comment]
Thanks a lot for the pointer to the migration guide! Not sure how I missed that 🤦‍♂️ (It might be useful to link to the migration guide in the release notes for discoverability.)

The main issue was the old usage of the `.extend` method as you pointed out. One additional aspect is that the extension of serializes also changed. I was initially thinking that it might be the same pattern as with `defaults` and so I created a `serializers` method that returned the extended serializers object. But that didn't work. I browsed through the source code and found out it's realized via a [static class variable](https://github.com/jupyter-widgets/ipywidgets/blob/940c10da0e724d110d13b87766a7b05bbe9c5e9d/packages/base/src/widget.ts#L688-L692). It might be useful to point that out in the migration guide as well.

Anyway, things seem to be working now 🎉 Thanks again for your help and pointers.
